llvm/lib/CodeGen/ReachingDefAnalysis.cpp

```diff
@@ -20,13 +20,55 @@ using namespace llvm;
 
 #define DEBUG_TYPE "reaching-defs-analysis"
 
-static cl::opt<bool> PrintAllReachingDefs("print-all-reaching-defs", cl::Hidden,
-                                          cl::desc("Used for test purpuses"),
-                                          cl::Hidden);
+AnalysisKey ReachingDefAnalysis::Key;
 
-char ReachingDefAnalysis::ID = 0;
-INITIALIZE_PASS(ReachingDefAnalysis, DEBUG_TYPE, "ReachingDefAnalysis", false,
-                true)
+ReachingDefAnalysis::Result
+ReachingDefAnalysis::run(MachineFunction &MF,
+                         MachineFunctionAnalysisManager &MFAM) {
+  ReachingDefInfo RDI;
```

Comment:
Use [MFPropsModifier](https://llvm.org/doxygen/classllvm_1_1MFPropsModifier.html#a0abbf442390713b95f5913ebe5987586) to handle properties, you can add getRequiredProperties method for this pass.